File: trunk/java/picocontainer/src/test/org/picocontainer/NoPicoTestCase.java
dust
/*****************************************************************************
 * Copyright (C) PicoContainer Organization. All rights reserved.            *
 * ------------------------------------------------------------------------- *
 * The software in this package is published under the terms of the BSD      *
 * style license a copy of which has been included with this distribution in *
 * the LICENSE.txt file.                                                     *
 *                                                                           *
 * Idea by Rachel Davies, Original code by Aslak Hellesoy and Paul Hammant   *
 *****************************************************************************/

package org.picocontainer;

import junit.framework.TestCase;
import org.picocontainer.testmodel.DependsOnTouchable;
import org.picocontainer.testmodel.SimpleTouchable;

public class NoPicoTestCase extends TestCase {

    /**
     * RecordingAware2 demonstration of using components WITHOUT Pico (or Nano)
     * This was one of the design goals.
     *
     * This is manual lacing of components.
     */
    public void testTouchableWithoutPicoTestCase() {

        SimpleTouchable touchable = new SimpleTouchable();
        new DependsOnTouchable(touchable);

        assertTrue("Touchable should have had its wasTouched method called",
                touchable.wasTouched);
    }
}

File: dotnet/picocontainer/src/Test/NoPicoTestCase.cs
/*****************************************************************************
 * Copyright (C) PicoContainer Organization. All rights reserved.            *
 * ------------------------------------------------------------------------- *
 * The software in this package is published under the terms of the BSD      *
 * style license a copy of which has been included with this distribution in *
 * the license.html file.                                                    *
 *                                                                           *
 * Idea by Rachel Davies, Original code by Aslak Hellesoy and Paul Hammant   *
 * C# port by Maarten Grootendorst                                           *
 *****************************************************************************/

using System;
using System.Diagnostics;

using csUnit;

using PicoContainer.Defaults;
using PicoContainer.Tests.TestModel;

namespace PicoContainer.Tests
{
	
  [TestFixture]
	public class NoPicoTestCase
	{
		
	 	public virtual void  testTouchableWithoutPicoTestCase()
		{
			
			SimpleTouchable touchable = new SimpleTouchable();
			new DependsOnTouchable(touchable);
			
			Assert.True(touchable._wasTouched,"Touchable should have had its wasTouched method called");
		}
	}
}